### DISCUSSION TITLE
Configure logging on standalone

### DISCUSSION THREAD

**Discussion by Samreay** [apache/pulsar#21350] [Q&A] [answered]
Hi all,

I've been struggling with getting the logging to be manageable when running `pulsar standalone` for unit testing purposes. By default, running a 

```bash
docker run -it -p 6650:6650 -p 8080:8080  --tmpfs /pulsar/data apachepulsar/pulsar:3.1.0 bin/pulsar standalone
```

Gives pages and pages of logs like this: 
![image](https://github.com/apache/pulsar/assets/6748505/bba08495-daa1-4964-b501-ab46b268fd34)

I haven't found any environment variables that work to control the logging at all, so right now our docker run has morphed into something that I'd consider... unsavoury.

```bash
docker run -it -p 6650:6650 -p 8080:8080  --tmpfs /pulsar/data apachepulsar/pulsar:3.1.0 /bin/bash -c "sed -i 's/info/error/g' /pulsar/conf/log4j2.yaml && sed -i 's/INFO/ERROR/g' /pulsar/conf/log4j2.yaml && sed -i 's/"\$\{sys:pulsar.log.level\}/error/g"' /pulsar/conf/log4j2.yaml && bin/pulsar standalone"
```

Which gives us something that doesn't make scrolling through logs when this runs in CI (and all the services are mixed together) a complete nightmare:

![image](https://github.com/apache/pulsar/assets/6748505/1706246a-2f30-454c-a786-4088f726d675)

But I presume there has to be a better way of silencing non-error logs than a bunch of `sed` commands. Does anyone know how to do this?

**Comment by jak78** [comment, marked as answer]
Hi!
You can use the `PULSAR_LOG_LEVEL` env variable:
```bash
docker run -it -p 6650:6650 -p 8080:8080 -e PULSAR_LOG_LEVEL=ERROR --tmpfs /pulsar/data apachepulsar/pulsar:3.1.0 bin/pulsar standalone
```

**Reply by Samreay** [reply]
Ah, I saw this issue https://github.com/apache/pulsar/issues/14298 asking for PULSAR_LOG_LEVEL and assumed that the issue being open meant that the env variable wasn't yet implemented. TIL. Thanks mate!

**Reply by jak78** [reply]
I'm happy your problem is solved. In my understanding, issue #14298 is about changing the log configuration at runtime without restarting the broker. This is still to be implemented.
